drivers/metadata/informationschema/metadata.go

```diff
@@ -821,6 +851,179 @@ func (s InformationSchema) Sequences(f metadata.Filter) (*metadata.SequenceSet,
+			"  ON t.table_catalog = tp.table_catalog AND t.table_schema = tp.table_schema AND t.table_name = tp.table_name"
+		qstrs = append(qstrs, qstr)
+	}
+
+	if s.hasColumnPrivileges {
+		columns := []string{
+			"t.table_catalog AS object_catalog",
+			"t.table_schema AS object_schema",
+			"t.table_name AS object_name",
+			"t.table_type AS object_type",
+			"column_name",
+			"grantee",
+			s.clauses[PrivilegesGrantor] + " AS grantor",
+			"privilege_type",
+			"CASE WHEN is_grantable='YES' THEN 1 ELSE 0 END AS is_grantable",
+		}
+
+		qstr := "SELECT\n" +
+			"  " + strings.Join(columns, ", ") + "\n" +
+			"FROM information_schema.column_privileges cp\n" +
+			"LEFT JOIN information_schema.tables t\n" +
+			"  ON t.table_catalog = cp.table_catalog AND t.table_schema = cp.table_schema AND t.table_name = cp.table_name"
+		qstrs = append(qstrs, qstr)
+	}
+
+	if s.hasUsagePrivileges {
+		columns := []string{
+			"object_catalog",
+			"object_schema",
+			"object_name",
+			"object_type",
+			"'' AS column_name",
+			"grantee",
+			s.clauses[PrivilegesGrantor] + " AS grantor",
+			"privilege_type",
+			"CASE WHEN is_grantable='YES' THEN 1 ELSE 0 END AS is_grantable",
+		}
+
+		qstr := "SELECT\n" +
+			"  " + strings.Join(columns, ", ") + "\n" +
+			"FROM information_schema.usage_privileges"
+		qstrs = append(qstrs, qstr)
+	}
+
+	// In the query result, table and column level privileges will be on separate rows.
+	// Each table or column can have multple privileges (i.e rows).
+	// For table level privileges the `column_name` column is empty.
+	qstr := "SELECT * FROM (\n" + strings.Join(qstrs, "\nUNION ALL\n") + "\n) AS subquery"
+	rows, closeRows, err := s.query(
+		qstr,
+		conds,
+		"object_catalog, object_schema, object_name, object_type, column_name, grantee, grantor, privilege_type",
+		vals...)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return metadata.NewPrivilegeSummarySet([]metadata.PrivilegeSummary{}), nil
+		}
+		return nil, err
+	}
+	defer closeRows()
+
+	type row struct {
+		Catalog       string
+		Schema        string
+		Name          string
+		ObjectType    string
+		Column        string
+		Grantee       string
+		Grantor       string
+		PrivilegeType string
+		IsGrantable   bool
+	}
+	// The rows need to be aggregated into one `metadata.PrivilegeSummary` object per table. The rows are ordered by table such that we can append
+	// to the current `metadata.PrivilegeSummary` as long as we are processing the same table.
+	results := []metadata.PrivilegeSummary{}
+	curSummary := &metadata.PrivilegeSummary{}
+	for rows.Next() {
+		r := row{}
+		err = rows.Scan(&r.Catalog, &r.Schema, &r.Name, &r.ObjectType, &r.Column, &r.Grantee, &r.Grantor, &r.PrivilegeType, &r.IsGrantable)
+		if err != nil {
+			return nil, err
+		}
+
+		switch {
+		// If the row specifies neither column nor table level privileges
+		case r.PrivilegeType == "":
+			if curSummary.Catalog != r.Catalog || curSummary.Schema != r.Schema || curSummary.Name != r.Name {
```

Comment:
Reverse the condition and do a `continue` to avoid one level of indentation.

Reply:
Not sure if I understood that. But I made it quite a bit simpler. Just without reversing the condition.